binary-oracle-pair/program/src/instruction.rs

```diff
@@ -0,0 +1,96 @@
+//! Instruction types
+
+//use crate::{
+//    error::PoolError,
+//};
+//use num_derive::{FromPrimitive, ToPrimitive};
+//use num_traits::{FromPrimitive, ToPrimitive};
+use solana_program::{
+    clock::Slot,
+    //sysvar,
+    //instruction::{AccountMeta},
+    program_error::ProgramError,
+    program_pack::{IsInitialized, Pack, Sealed},
+};
+
+//use std::{convert::TryInto, mem::size_of};
+
+/// Instruction definition
+#[derive(Clone, Debug, PartialEq)]
+pub enum Instruction {
+    /// Initializes a new binary oracle pair pool.
+    ///
+    ///   0. `[writable]` Pool account.
+    ///   1. `[]` authority create_program_address(&[Token-swap account])`
+    ///   2. `[]` Decider authority
+    ///   3. `[]` Deposit currency SPL Token mint. Must be initialized.
```

Comment:
Small point: we'll want to prevent any further deposits on the token swap

Reply:
we just need to prevent minting new bops

Reply:
Right, you can strike this one, I thought a token swap was tied to the BOP